src/impl/tpls/KokkosBlas3_gemm_tpl_spec_decl.hpp

```diff
@@ -817,4 +380,166 @@ KOKKOSBLAS3_CGEMM_CUBLAS(Kokkos::LayoutRight, Kokkos::LayoutRight,
+      Kokkos::Profiling::pushRegion(                                         \
+          "KokkosBlas::gemm[TPL_ROCBLAS," #SCALAR_TYPE "]");                 \
+                                                                             \
+      const bool A_t = (transA[0] != 'N') && (transA[0] != 'n');             \
+      const int M    = static_cast<int>(C.extent(0));                        \
+      const int N    = static_cast<int>(C.extent(1));                        \
+      const int K    = static_cast<int>(A.extent(A_t ? 0 : 1));              \
+                                                                             \
+      bool is_lr = std::is_same<Kokkos::LayoutRight, LAYOUT>::value;         \
+                                                                             \
+      const int AST = is_lr ? A.stride(0) : A.stride(1),                     \
+                LDA = AST == 0 ? 1 : AST;                                    \
+      const int BST = is_lr ? B.stride(0) : B.stride(1),                     \
+                LDB = BST == 0 ? 1 : BST;                                    \
+      const int CST = is_lr ? C.stride(0) : C.stride(1),                     \
+                LDC = CST == 0 ? 1 : CST;                                    \
+                                                                             \
+      rocblas_operation transa = trans_mode_kk_to_rocblas(transA);           \
+      rocblas_operation transb = trans_mode_kk_to_rocblas(transB);           \
+                                                                             \
+      constexpr int numDotsLayoutLeftThreshold  = 1600;                      \
+      constexpr int numDotsLayoutRightThreshold = 100;                       \
+      if ((!is_lr && transa != rocblas_operation_none &&                     \
+           transb == rocblas_operation_none &&                               \
+           M * N < numDotsLayoutLeftThreshold) ||                            \
+          (is_lr && transa != rocblas_operation_none &&                      \
+           transb == rocblas_operation_none &&                               \
+           M * N < numDotsLayoutRightThreshold)) {                           \
+        DotBasedGEMM<ExecSpace, AViewType, BViewType, CViewType> gemm(       \
+            alpha, A, B, beta, C);                                           \
+        bool conjT =                                                         \
+            (std::is_same<SCALAR, double>::value ||                          \
+             std::is_same<SCALAR, float>::value)                             \
+                ? false                                                      \
+                : (transa == rocblas_operation_conjugate_transpose ? true    \
+                                                                   : false); \
+        gemm.run(space, conjT);                                              \
```

Comment:
Just confirming - you want to call this KokkosKernels gemm algorithm on AMD GPUs like we do for TPL functions on CUDA GPUs?

Reply:
@e10harvey Yes, for now I just follow what the cuBLAS wrapper does. So I assume the same thresholds apply here for DotBasedGemm. We need to test the performance more carefully for this case but I do not have time now.